pkg/kubernetes/manifests.go

```diff
@@ -37,19 +37,35 @@ func NewRawExtensions(manifest []byte) ([]runtime.RawExtension, error) {
 
 // NewList decoded data into a list of Kubernetes resources
 func NewList(data []byte) (*metav1.List, error) {
-	list := metav1.List{}
+	return parseList(data, false)
+}
+
+// ParseListLax is like NewList but ignores comment and whitespace nodes
+func ParseListLax(data []byte) (*metav1.List, error) {
+	return parseList(data, true)
+}
```

Comment:
how do we know when to call this vs `NewList`? Shouldn't we just always ignore comments in every yaml resource we parse?

Reply:
We'd want to use `ParseListLax` for YAMLs obtained from external sources. 

>Shouldn't we just always ignore comments in every yaml resource we parse?

Probably. I kept the implementation of NewList unchanged because I didn't want to change the existing behaviour for its callers.

Reply:
> We'd want to use `ParseListLax` for YAMLs obtained from external sources.

shall we change the name of the function to represent that then? Wouldn't really of known otherwise :see_no_evil: 

> > Shouldn't we just always ignore comments in every yaml resource we parse?
> 
> Probably. I kept the implementation of NewList unchanged because I didn't want to change the existing behaviour for its callers.

We could just make it the default behaviour :shrug: don't see any harm

Reply:
>shall we change the name of the function to represent that then? Wouldn't really of known otherwise see_no_evil

What are you thinking of?

>We could just make it the default behaviour shrug don't see any harm

I was worried about breaking existing code as NewList is being used in a few places, but I'll go ahead with making it the default behaviour and run the integration tests on this branch before merging.

Reply:
> > shall we change the name of the function to represent that then? Wouldn't really of known otherwise see_no_evil
> 
> What are you thinking of?
> > We could just make it the default behaviour shrug don't see any harm
> 
> I was worried about breaking existing code as NewList is being used in a few places, but I'll go ahead with making it the default behaviour and run the integration tests on this branch before merging.

If its going to be the default behaviour we don't need to rename it :smile: but if we do end up keeping them separate somethings more descriptive e.g. `ParseKubernetesResourcesWithComments`. I just find `ParseListLax` means nothing to me :joy:

Reply:
>If its going to be the default behaviour we don't need to rename it

Yup, we don't. 

>somethings more descriptive e.g. ParseKubernetesResourcesWithComments

How about `ParseResourcesIgnoreComments`?

Reply:
sounds good, although hopefully it will just be the default behaviour :crossed_fingers: